src/hash.cr

```diff
@@ -23,10 +24,36 @@ class Hash(K, V)
     @block = block
   end
 
+  # Creates a new empty `Hash` of *initial_capacity* with a *block* that handles missing keys.
+  #
+  # The *initial_capacity* is useful to avoid unnecessary reallocations
+  # of the internal buffer in case of growth. If you have an estimate
```

Comment:
I just copied and pasted this from the `Array` docs. Not totally sure it's entirely accurate here since I know that bucket size also is used for determining when to rehash. Should I mostly keep it as is but change the voice to avoid "I, me, you" stuff? Should I update the `Array` docs while I'm at it?